modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/access/token/grant/handler/LiferayResourceOwnerAccessTokenGrantHandler.java

```diff
@@ -104,40 +104,22 @@ protected boolean isGrantHandlerEnabled() {
 			allowResourceOwnerPasswordCredentialsGrant();
 	}
 
+	private static final Log _log = LogFactoryUtil.getLog(
+		LiferayResourceOwnerAccessTokenGrantHandler.class);
+
 	@Reference(
 		policy = ReferencePolicy.DYNAMIC,
 		policyOption = ReferencePolicyOption.GREEDY
 	)
-	protected void setLiferayOAuthDataProvider(
-		LiferayOAuthDataProvider liferayOAuthDataProvider) {
+	private volatile LiferayOAuthDataProvider _liferayOAuthDataProvider;
```

Comment:
@W2jx This does not need to be dynaimc and greedy, as LiferayOAuthDataProvider is an internal class.